### PR TITLE
Fix #612 - Port channels log from stderr to logging framework

### LINKED ISSUE #612: runserver should log to "django.channels.server" instead of stderr
Django v1.10 switched to using "django.server" as the logger for runserver. My console is being spammed with stderr messages that I can't configure in development when using channels, and it's very difficult to see what I want to see, which is not the 200 requests. If you would please, follow suit with Django and set up a logger using "django.channels.server" and change the runserver stderr logging to use that logger instead. Even if it doesn't follow the 200 = INFO, 400+ = ERROR paradigm, it would be nice to be able to shut that off so I can see what I'm looking for in other dev logs.

### PR DESCRIPTION
**Pull request by renatooliveira**
#612 

`channels/log.py` is basically a copy of django log itself.

Some decisions I made here that I don't know if they're right:

1 - I moved the logger object away from the class because it was being called before instantiation
2 - I removed formatters from DEFAULT_LOGGING due to backwards compatibility (It was only added on 1.10)

I'm also not sure if the tests are right, and would like some guidance on this.

Looking forward to hear your feedback!

Thanks!

### CODE REVIEW

**Comment by renatooliveira** [comment]
I just saw that the checks failed, I'll fix it.

**Comment by renatooliveira** [comment]
hey @andrewgodwin is there anything I can do to help with this PR?

**Comment by proofit404** [comment]
I'll review it during this week. Thanks for your contribution!

**Comment by renatooliveira** [comment]
Thanks @proofit404 I'll do the changes this week and let you know

File: channels/log.py [hunk 1]
```diff
@@ -1,31 +1,88 @@
...
+        'django': {
+            'handlers': ['console', 'mail_admins'],
+            'level': 'INFO',
+        },
+        'django.server': {
+            'handlers': ['django.server'],
+            'level': 'INFO',
+            'propagate': False,
+        },
+        'django.channels.server': {
```
Comment (renatooliveira) [inline channels/log.py]: Hi @proofit404 what should I use here? I used what was proposed on the github issue.

Thanks

Comment (proofit404) [inline channels/log.py]: I'll reread related issue and leave a more helpful comment.

**Comment by proofit404** [comment]
Since channels package isn't part of the Django code base, I don't think its good idea to use `django.channels.server` as the logger name.

Could you answer what behavior we expect, if someone run development server with `--use-asgi=false`?

Also, can you write documentation chapter on configuration logging with changes you've done?

**Comment by renatooliveira** [comment]
Hey @proofit404, I changed from `django.channels.server` to `channels.server`.

Regarding `--use-asgi=false`, you mean `--noasgi`, right? I'm not sure which behavior I should expect on this. I tested against master and my branch and both worked the same way.

I'll start writing the docs for this

**Comment by jjlorenzo** [comment]
any updates?

**Comment by proofit404** [comment]
I'll return to this PR at the weekend.

**Comment by proofit404** [comment]
I've tried this PR. Looks like it working. But I need more time to review it intently. Sorry for the delay.

**Comment by slampenny** [comment]
Really looking forward to this merge. It's been very hard to debug without access to the log.

**Comment by amir-hadi** [comment]
One side question: Why would you set propagate to False on the loggers? The old implementation breaks up my logging config: It sets propagate = True, even if I configure it on my own in the global LOGGING (dictConfig) config. I believe this change would respect the global configuration of the logging better, so if I set propagate = True on the channels logger, it will stay like that.

The reason why I am asking: From my understanding when you set propagate to False the log messages don't get forwarded to the root logger and so global handlers are not invoked. I have a SentryHandler on the root logger, which means every error or exception gets reported to Sentry (which is desired). So why would one set propagate = False as default?

**Comment by andrewgodwin** [comment]
This PR is now super mismatched to the current release so I'm going to close it, but I would still like to see logging improved somewhat (though I did some improvements in Channels 2 anyway)

**Pull request closed without merging**